cmd/goal/messages.go

```diff
@@ -156,6 +156,9 @@ const (
 	tealsignTooManyArg    = "--set-lsig-arg-idx too large, maximum of %d arguments"
 	tealsignInfoWroteSig  = "Wrote signature for %s to LSig.Args[%d]"
 
+	tealLogicSigSize = "%s: logicsig program size too large: %d > %d"
```

Comment:
I dislike the messages.go pattern; the strings are only used once and make more sense in context. The other two added reportErrorf() calls in clerk.go have inline strings.

Reply:
I somewhat agree, but you and @tsachiherman should arm wrestle and tell me who wins.  I don't feel too strongly about it.

Reply:
I know that it's a bit of a pain to maintain it *right now*, but the separation between code and text is meant to help us with few things ( even if we don't get the gains right now ) : 
- avoiding duplicate strings
- correcting strings ( typo, or just something more systematic such as the usage of the word "Algo" )
- Support for localization
- Support for non-ASCII output characters on different platforms.

Reply:
I used to believe the i18n benefit, and still kinda do, but also saw a system with effectively `func i18n(string) string` so that at a call site you could `fmt.Printf(i18n("Hello World, %s"), foo)` and it might substitute in "Hola Mundo, %s" based on a translation table lookup.
Anyway, I don't feel strong enough about it to block this PR. Might be a topic for `#dev` slack discussion to get a sense of people's feelings on style.